Change to IpGroups causes crash Sinds version 3.75.0

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.6.4

### AzureRM Provider Version

azurerm_v3.79.0_x5

### Affected Resource(s)/Data Source(s)

azurerm_ip_group, azurerm_firewall_policies

### Terraform Configuration Files

```hcl
ip_groups = {
  G-XXXXXL-NETWORKS = {
    cidrs = [
      "xxx.xxx.xxx.0/22",
    ]
    name               = "G-XXXXXXX-NETWORKS"
    resource_group_key = "hub-xx-weu-networxking-ipgxros"
    tags = {
      "environment" : "Integration"
    }
  }
}
```


### Debug Output/Panic Output

```shell
Error: Plugin did not respond
│ 
│   with module.caf.module.azurerm_firewall_policies["XXX-XXX-XX"].azurerm_firewall_policy.fwpol,
│   on .terraform/modules/caf/modules/networking/firewall_policies/firewall_policy.tf line 16, in resource "azurerm_firewall_policy" "fwpol":
│   16: resource "azurerm_firewall_policy" "fwpol" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.caf.module.ip_groups["XXX-XXX-XXX"].azurerm_ip_group.ip_group,
│   on .terraform/modules/caf/modules/networking/ip_group/module.tf line 24, in resource "azurerm_ip_group" "ip_group":
│   24: resource "azurerm_ip_group" "ip_group" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v3.79.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x10a74a0a8]

goroutine 125 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.resourceIpGroupUpdate(0x0?, {0x10c9329a0?, 0x14001c23680?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/ip_group_resource.go:206 +0x2d8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x10d804f48?, {0x10d804f48?, 0x1400280f1d0?}, 0xd?, {0x10c9329a0?, 0x14001c23680?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:800 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14001496fc0, {0x10d804f48, 0x1400280f1d0}, 0x14000ed3a00, 0x140024aa100, {0x10c9329a0, 0x14001c23680})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x658
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140004cb248, {0x10d804f48?, 0x1400280f0e0?}, 0x1400202f2c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xb08
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140012906e0, {0x10d804f48?, 0x1400280e8d0?}, 0x1400025f340)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10d410c20?, 0x140012906e0}, {0x10d804f48, 0x1400280e8d0}, 0x1400025f2d0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002281e0, {0x10d8166c0, 0x140004051e0}, 0x1400280ca20, 0x14001504e40, 0x1132eac78, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1374 +0xb44
google.golang.org/grpc.(*Server).handleStream(0x140002281e0, {0x10d8166c0, 0x140004051e0}, 0x1400280ca20, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1751 +0x7c4
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.58.3/server.go:986 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 41
        google.golang.org/grpc@v1.58.3/server.go:997 +0x160

Error: The terraform-provider-azurerm_v3.79.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

An correct update of the I-groups

### Actual Behaviour

Stack trace from the terraform-provider-azurerm_v3.79.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x10a74a0a8]

goroutine 125 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.resourceIpGroupUpdate(0x0?, {0x10c9329a0?, 0x14001c23680?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/ip_group_resource.go:206 +0x2d8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x10d804f48?, {0x10d804f48?, 0x1400280f1d0?}, 0xd?, {0x10c9329a0?, 0x14001c23680?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:800 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14001496fc0, {0x10d804f48, 0x1400280f1d0}, 0x14000ed3a00, 0x140024aa100, {0x10c9329a0, 0x14001c23680})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:919 +0x658
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140004cb248, {0x10d804f48?, 0x1400280f0e0?}, 0x1400202f2c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xb08
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140012906e0, {0x10d804f48?, 0x1400280e8d0?}, 0x1400025f340)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10d410c20?, 0x140012906e0}, {0x10d804f48, 0x1400280e8d0}, 0x1400025f2d0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002281e0, {0x10d8166c0, 0x140004051e0}, 0x1400280ca20, 0x14001504e40, 0x1132eac78, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1374 +0xb44
google.golang.org/grpc.(*Server).handleStream(0x140002281e0, {0x10d8166c0, 0x140004051e0}, 0x1400280ca20, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1751 +0x7c4
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.58.3/server.go:986 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 41
        google.golang.org/grpc@v1.58.3/server.go:997 +0x160

Error: The terraform-provider-azurerm_v3.79.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Steps to Reproduce

I have traced the issue back to release 3.75.0. Version 3.74.0 was still working correctly, but as of version 3.75.0 I see this 

ENHANCEMENTS:

dependencies: firewall migrated to hashicorp/go-azure-sdk (https://github.com/hashicorp/terraform-provider-azurerm/issues/22863) 

Everything is working correctly with version 3.74.0, the crashes start happening at version 3.75.0

### Important Factoids

Everything is working correctly with version 3.74.0, the crashes start happening at version 3.75.0

### References

_No response_

--- a/internal/services/network/ip_group_resource.go
+++ b/internal/services/network/ip_group_resource.go
@@ -183,8 +183,25 @@ func resourceIpGroupRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		}
 	}
 
-	d.Set("firewall_ids", getIds(resp.Firewalls))
-	d.Set("firewall_policy_ids", getIds(resp.FirewallPolicies))
+	var firewallIDs []string
+	for _, idStr := range getIds(resp.Firewalls) {
+		firewallID, err := azurefirewalls.ParseAzureFirewallIDInsensitively(idStr)
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall ID %q: %+v", idStr, err)
+		}
+		firewallIDs = append(firewallIDs, firewallID.ID())
+	}
+	d.Set("firewall_ids", firewallIDs)
+
+	var firewallPolicyIDs []string
+	for _, idStr := range getIds(resp.FirewallPolicies) {
+		policyID, err := firewallpolicies.ParseFirewallPolicyIDInsensitively(idStr)
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall Policy ID %q: %+v", idStr, err)
+		}
+		firewallPolicyIDs = append(firewallPolicyIDs, policyID.ID())
+	}
+	d.Set("firewall_policy_ids", firewallPolicyIDs)
 
 	return tags.FlattenAndSet(d, resp.Tags)
 }

--- a/internal/services/network/ip_group_resource.go
+++ b/internal/services/network/ip_group_resource.go
@@ -8,6 +8,7 @@ import (
 	"log"
 	"time"
 
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2023-06-01/azurefirewalls"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2023-06-01/firewallpolicies"
@@ -93,13 +94,19 @@ func resourceIpGroupCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 	defer cancel()
 
 	for _, fw := range d.Get("firewall_ids").([]interface{}) {
-		id, _ := azurefirewalls.ParseAzureFirewallID(fw.(string))
+		id, err := azurefirewalls.ParseAzureFirewallID(fw.(string))
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall ID %q: %+v", fw, err)
+		}
 		locks.ByName(id.AzureFirewallName, firewall.AzureFirewallResourceName)
 		defer locks.UnlockByName(id.AzureFirewallName, firewall.AzureFirewallResourceName)
 	}
 
 	for _, fwpol := range d.Get("firewall_policy_ids").([]interface{}) {
-		id, _ := firewallpolicies.ParseFirewallPolicyID(fwpol.(string))
+		id, err := firewallpolicies.ParseFirewallPolicyID(fwpol.(string))
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall Policy ID %q: %+v", fwpol, err)
+		}
 		locks.ByName(id.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
 		defer locks.UnlockByName(id.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
 	}
@@ -183,7 +190,7 @@ func resourceIpGroupRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		}
 	}
 
-	var firewallIDs []string
+	firewallIDs := make([]string, 0)
 	for _, idStr := range getIds(resp.Firewalls) {
 		firewallID, err := azurefirewalls.ParseAzureFirewallIDInsensitively(idStr)
 		if err != nil {
@@ -193,7 +200,7 @@ func resourceIpGroupRead(d *pluginsdk.ResourceData, meta interface{}) error {
 	}
 	d.Set("firewall_ids", firewallIDs)
 
-	var firewallPolicyIDs []string
+	firewallPolicyIDs := make([]string, 0)
 	for _, idStr := range getIds(resp.FirewallPolicies) {
 		policyID, err := firewallpolicies.ParseFirewallPolicyIDInsensitively(idStr)
 		if err != nil {
@@ -213,13 +220,19 @@ func resourceIpGroupUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 	defer cancel()
 
 	for _, fw := range d.Get("firewall_ids").([]interface{}) {
-		id, _ := azurefirewalls.ParseAzureFirewallID(fw.(string))
+		id, err := azurefirewalls.ParseAzureFirewallID(fw.(string))
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall ID %q: %+v", fw, err)
+		}
 		locks.ByName(id.AzureFirewallName, firewall.AzureFirewallResourceName)
 		defer locks.UnlockByName(id.AzureFirewallName, firewall.AzureFirewallResourceName)
 	}
 
 	for _, fwpol := range d.Get("firewall_policy_ids").([]interface{}) {
-		id, _ := firewallpolicies.ParseFirewallPolicyID(fwpol.(string))
+		id, err := firewallpolicies.ParseFirewallPolicyID(fwpol.(string))
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall Policy ID %q: %+v", fwpol, err)
+		}
 		locks.ByName(id.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
 		defer locks.UnlockByName(id.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
 	}
@@ -304,15 +317,21 @@ func resourceIpGroupDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	}
 
 	for _, fw := range *read.Firewalls {
-		id, _ := azurefirewalls.ParseAzureFirewallID(*fw.ID)
-		locks.ByName(id.AzureFirewallName, firewall.AzureFirewallResourceName)
-		defer locks.UnlockByName(id.AzureFirewallName, firewall.AzureFirewallResourceName)
+		fwID, err := azurefirewalls.ParseAzureFirewallID(pointer.From(fw.ID))
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall ID %q: %+v", pointer.From(fw.ID), err)
+		}
+		locks.ByName(fwID.AzureFirewallName, firewall.AzureFirewallResourceName)
+		defer locks.UnlockByName(fwID.AzureFirewallName, firewall.AzureFirewallResourceName)
 	}
 
 	for _, fwpol := range *read.FirewallPolicies {
-		id, _ := firewallpolicies.ParseFirewallPolicyID(*fwpol.ID)
-		locks.ByName(id.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
-		defer locks.UnlockByName(id.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
+		polID, err := firewallpolicies.ParseFirewallPolicyID(pointer.From(fwpol.ID))
+		if err != nil {
+			return fmt.Errorf("parsing Azure Firewall Policy ID %q: %+v", *fwpol.ID, err)
+		}
+		locks.ByName(polID.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
+		defer locks.UnlockByName(polID.FirewallPolicyName, firewall.AzureFirewallPolicyResourceName)
 	}
 
 	future, err := client.Delete(ctx, id.ResourceGroup, id.Name)

